app/controllers/questions/eip_overview_controller.rb

```diff
@@ -21,7 +21,7 @@ def after_update_success
       session[:intake_id] = @form.intake.id
       stimulus_triage_id = session.delete(:stimulus_triage_id)
       if stimulus_triage_id.present?
-        current_intake.update(triage_source: StimulusTriage.find(stimulus_triage_id), visitor_id: visitor_id)
+        @form.intake.update(triage_source: StimulusTriage.find(stimulus_triage_id))
```

Comment:
another thing that could be done to fix this is the `current_intake` method on this controller could return `super || Intake.new(eip_only: true)` which would also take care of people who arrived on this page with an intake already in their session? unclear whether this is what we want for people arriving on the page or not (unclear duplicate behavior etc etc)

Reply:
It seems like the logic was intended to always create a new intake for EIP only, even if they already had another intake. So super || would fix the after_update issue, but would cause a bug on the before update. I think this fix is the "least tricky" way to ensure that we're actually doing what we want, which is updating the immediately-previously-created intake with this new data.